### DISCUSSION TITLE
Service user wants to have AWS Security Hub findings state updated based on the latest run of kube-bench

### DISCUSSION THREAD

**Discussion by build-failure** [aquasecurity/kube-bench#1267] [Development]
# Description
As a service user I want the status of existing security hub findings being updated based on the latest run of kube-bench.

# Acceptance Criteria
- Findings for checks with previous state *FAIL* or *WARN* that now have state *PASS* are updated to status *RESOLVED* within AWS Security Hub.
- Findings for checks with previous state *PASS* that now have state *FAIL* or *WARN* are updated to status *NEW* within AWS Security Hub.

# Remarks
IMHO based on the related [controls.go code](https://github.com/aquasecurity/kube-bench/blob/main/check/controls.go#L230), ATM only checks with the state *FAIL* and *WARN* are [fetched](https://github.com/aquasecurity/kube-bench/blob/main/internal/findings/publisher.go#L54) into AWS Security Hub using[ BatchImportFindings API](https://docs.aws.amazon.com/securityhub/1.0/APIReference/API_BatchImportFindings.html) call.

**Comment by mozillazg** [comment]
IMHO, this case is not the work scope of kube-bench. Maybe creating a new project which accepts output from kube-bench to handle this case is better?
